pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -184,7 +184,8 @@ func (e *EndpointController) getPodServiceMemberships(pod *api.Pod) (sets.String
 	for i := range services {
 		key, err := keyFunc(&services[i])
```

Comment:
I think if here return error,  addPod or updatePod func will return directly, it can't force a requeue for pod, like below

	if err != nil {
		glog.Errorf("Unable to get pod %v/%v's service memberships: %v", pod.Namespace, pod.Name, err)
		return
	}



then, the pod related service still doesn't get synced